### PR TITLE
nixos/tests: drop LegacyStartCommand

### PR DESCRIPTION
**Pull request by K900**
## Description of changes

Switches the remaining tests that use the `LegacyStartCommand` API to generate a start command in the test body and pass it to the driver.

All the affected tests are still working (at least on x86_64) after this change, except `boot.ubootExtlinux`, which hasn't worked before this change either.

The commits are roughly ordered as least objectionable to most objectionable, so if people really want to keep the existing API, I can move that to a separate change.

Also, gave all the installer/boot tests more RAM, because they are otherwise hitting swap heavily, which slows things down A LOT.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/virtualization/xen/4.15.nix [hunk 1]
```diff
@@ -125,7 +125,7 @@ callPackage (import ./generic.nix (rec {
     ++ optional (withSeabios) "--with-system-seabios=${seabios}/share/seabios"
     ++ optional (!withInternalSeabios && !withSeabios) "--disable-seabios"
 
-    ++ optional (withOVMF) "--with-system-ovmf=${OVMF.fd}/FV/OVMF.fd"
+    ++ optional (withOVMF) "--with-system-ovmf=${OVMF.firmware}"
```
Comment (roberth) [inline pkgs/applications/virtualization/xen/4.15.nix:128]: This is a different file.
```
nix-repl> OVMF.firmware
"/nix/store/jhnng8aqgw824a08ly9h878rzzgm5m92-OVMF-202402-fd/FV/OVMF_CODE.fd"
```

The other day I tried to boot a USB stick in `qemu`, but it didn't work with this attribute; I had to use `OVMF.fd`, not `OVMF_CODE.fd`.

Maybe this use case is different; I'm certainly no expert on this, but it looks wrong.

Comment (K900) [reply pkgs/applications/virtualization/xen/4.15.nix:128]: You were probably using `-bios OVMF.fd`, which is not entirely correct, you actually need to use the `OVMF_CODE.fd` and `OVMF_VARS.fd` files separately. AFAICT Xen should do the right thing here, though I haven't tested it directly.

**Comment by adamcstephens** [comment]
@K900  do you think the dropping of OVMF.fd is too much of a breaking change that we should roll all or part of https://github.com/NixOS/nixpkgs/pull/284874 back?

**Comment by K900** [comment]
> @K900 do you think the dropping of OVMF.fd is too much of a breaking change that we should roll all or part of #284874 back?

The changes here are OK I think, there aren't that many direct dependencies for OVMF.

**Comment by K900** [comment]
I think we should apply your fix first then, and we can merge this later when there's no rush.

**Comment by shlevy** [comment]
https://github.com/NixOS/nixpkgs/pull/291565

File: pkgs/tools/virtualization/multipass/default.nix [hunk 2]
```diff
@@ -70,8 +70,8 @@ stdenv.mkDerivation
 
     # Patch the patch of the OVMF binaries to use paths from the nix store.
     substituteInPlace ./src/platform/backends/qemu/linux/qemu_platform_detail_linux.cpp \
-      --replace "OVMF.fd" "${OVMF.fd}/FV/OVMF.fd" \
-      --replace "QEMU_EFI.fd" "${OVMF.fd}/FV/QEMU_EFI.fd"
+      --replace "OVMF.fd" "${OVMF.firmware}" \
```
Comment (shlevy) [inline pkgs/tools/virtualization/multipass/default.nix:73]: Have you checked how `multipass` is using these variables?

Comment (K900) [reply pkgs/tools/virtualization/multipass/default.nix:73]: https://github.com/canonical/multipass/blob/f51d0ccf83b42e1e6a64fbfa8de9b7602d45192e/src/platform/backends/qemu/linux/qemu_platform_detail_linux.cpp#L175-L184

_oh no_

Comment (K900) [reply pkgs/tools/virtualization/multipass/default.nix]: OK I guess we can drop that change.

**Comment by K900** [comment]
This PR is just scoped to tests now.

**Pull request merged**